src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -243,23 +243,66 @@ private void FindJsonFilesInDirectoryRecursive(DirectoryInfo directoryPath, List
+        private static bool IsValidTemplate(string template)
         {
-            JObject jsonTemplate = JObject.Parse(template);
-            string schema = (string)jsonTemplate["$schema"];
-            string[] validSchemas = { 
-                "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"};
-            return validSchemas.Contains(schema);
+            var reader = new JsonTextReader(new StringReader(template));
+
+            reader.Read();
+            if (reader.TokenType != JsonToken.StartObject)
+            {
+                return false;
+            }
+
+            // Checks all top level properties for schema property. If schema is not found, will check property names against expected ARM template properties. If expected properties
+            // are not found function will return false.
+            while (reader.Read())
+            {
+                if (reader.Depth == 1 && reader.TokenType == JsonToken.PropertyName)
+                {
+                    string[] expectedProperties =
+                    {
+                        "contentVersion",
+                        "apiProfile",
+                        "parameters",
+                        "variables",
+                        "functions",
+                        "resources",
+                        "outputs",
+                    };
+
+                    if (reader.Value.ToString() == "$schema")
+                    {
+                        reader.Read();
+
+                        if (reader.TokenType != JsonToken.String)
+                        {
+                            return false;
+                        }
+
+                        string[] validSchemas = {
+                            "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                            "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                            "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
+                            "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#",
+                            "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"
+                        };
+
+                        return validSchemas.Contains(reader.Value);
+                    }
+                    else if (!expectedProperties.Contains(reader.Value))
```

Comment:
ARM templates are not case-sensitive.  So if a template has a resources array with the key `Resources` instead of `resources`, this check will fail.  I suggest you do a case-insensitive comparison.
```suggestion
                    else if (!expectedProperties.Any(property => property.Equals(reader.Value, StringComparison.OrdinalIgnoreCase)))
```